modules/core/test/src/smithy4s/http/internals/MetadataSpec.scala

```diff
@@ -160,7 +161,8 @@ class MetadataSpec() extends FunSuite {
   // QUERY PARAMETERS
   // ///////////////////////////////////////////////////////////
   test("String query parameter") {
-    val queries = Queries(str = Some("hello"))
+    val queries =
+      Queries(str = Some("hello"), slm = Some(Map("str" -> "hello")))
```

Comment:
I'd like for the values to differ so that the precedence can be exhibited

Reply:
Also, rather than modifying the existing tests, I'd rather you created a new one with a name reflecting what we're actually testing for.

Reply:
Okay, in that case I'm not certain I agree with the semantics proposed by AWS (in the context of our protocol anyway), because round-trip invariants are a very important notion in serialisation/deserialisation. 

> which is that all query params should end up in the structure that is annotated with the httpQueryParams regardless if they are also bound to another structure member

Can you link to either the spec that says so or the test you inferred it from so that I can have a look and cry ?

Reply:
Yes, this notion spun me up ,when i saw it..
Sure , please livestream the crying
Here are 2 request tests which applies to both server and client
https://github.com/awslabs/smithy/blob/8f32289b8f23bd53b1676af34f703680a79c6daa/smithy-aws-protocol-tests/model/restJson1/http-query.smithy#L33

https://github.com/awslabs/smithy/blob/8f32289b8f23bd53b1676af34f703680a79c6daa/smithy-aws-protocol-tests/model/restJson1/http-query.smithy#L145

Reply:
I think diverging with the AWS specs here wouldn't make much sense, even if round-tripping is violated. 

I agree to roll with your implementation. 

**However**, the tests need to be reworked. The tests were initially written based on the round-tripping invariant holding. Now that it doesn't hold anymore, you to change it to something like : 

```scala
def checkRoundTrip(initial: Queries, encoded: Metadata, final: Queries)  
```

because we need to test that `Queries(int = Some(123)` becomes `Queries(int = Some(123), slm = Some(Map("int" -> 123))` 

Reply:
Makes sense